### PR TITLE
Send package published notification to package owner(s) when new version is published

### PR DESCRIPTION
**Pull request by maartenba**
Send package published notification to package owner(s) when new version
is published

### CODE REVIEW

**Comment by maartenba** [comment]
@skofman1 @xavierdecoster

File: src/NuGetGallery/Controllers/PackagesController.cs [hunk 1]
```diff
@@ -1106,6 +1106,11 @@ private ActionResult EditFailed(string id, EditPackageRequest formData, Package
 
                 // tell Lucene to update index for the new package
                 _indexingService.UpdateIndex();
+
+                _messageService.SendPackageAddedNotice(package,
```
Comment (skofman1) [inline src/NuGetGallery/Controllers/PackagesController.cs]: this method is called from the website only. Do we want to send an e-mail in this scenario as well? If the user is here he is already authenticated, so is e-mail necessary?

Comment (maartenba) [inline src/NuGetGallery/Controllers/PackagesController.cs]: Good question. In theory someone could script a form upload as well.

@yishaigalatzer / @csharpfritz thoughts?

Comment (csharpfritz) [inline src/NuGetGallery/Controllers/PackagesController.cs]: As a security upgrade, I like adding an email notification to all package owners.  I'd like to see this enabled by default, but configurable as a preference on my NuGet profile page

Comment (maartenba) [inline src/NuGetGallery/Controllers/PackagesController.cs]: But only for API uploads or also web uploads?

Comment (yishaigalatzer) [inline src/NuGetGallery/Controllers/PackagesController.cs]: all

**Pull request merged**